streams/src/main/java/org/apache/kafka/streams/Topology.java

```diff
@@ -119,10 +119,14 @@ private static AutoOffsetResetInternal convertOldToNew(final Topology.AutoOffset
      * @return itself
      *
      * @throws TopologyException
-     *         if the provided source name is not unique, or
-     *         if topics have already been registered by another source,
+     *         if the provided source name is not unique,
+     *         no topics are specified, or
+     *         a topic has already been registered by another source,
      *         {@link #addReadOnlyStateStore(StoreBuilder, String, Deserializer, Deserializer, String, String, ProcessorSupplier) read-only state store}, or
      *         {@link #addGlobalStore(StoreBuilder, String, Deserializer, Deserializer, String, String, ProcessorSupplier) global state store}
+     * @throws NullPointerException
+     *         if {@code name} or {@code topics} is {@code null}, or
+     *         {@code topics} contains a {@code null} topic
```

Comment:
Not sure if we need/want do document `NullPointerException`? Could be too verbose. Added if for now, but also ok if we want to remove it again.

While I was looking into it, I actually did find some stuff in `InternalTopologyBuilder` that was not right and I tried to fix it...